backend/capellacollab/projects/toolmodels/diagrams/routes.py

```diff
@@ -52,7 +52,7 @@ async def get_diagram_metadata(
     except requests.exceptions.HTTPError:
         logger.info("Failed fetching diagram metadata", exc_info=True)
         raise fastapi.HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            status_code=status.HTTP_404_NOT_FOUND,
```

Comment:
I think `HTTP_422_UNPROCESSABLE_ENTITY` would be better here. We can't say if the resource doesn't exist or if it's another error. The case is more or less the wildcard for all errors.